src/api/routes/synapse/room_events.rs

```diff
@@ -68,7 +68,7 @@ lazy_static::lazy_static! {
 }
 
 impl FriendshipEvent {
-    fn validate_new_event_is_valid(
+    pub fn validate_new_event_is_valid(
```

Comment:
If we are using this method in `ws` also, then we should think of moving this out to a shared folder something like `ports/friendship_synapse`? In the way that `ports/users_cache.rs` is